ci/pkl-pipelines/pay-deploy/infra-drift-detector.pkl

```diff
@@ -0,0 +1,153 @@
+      new TaskStep {
+        task = "check-for-drift"
+        file = "pay-ci/ci/tasks/check-for-tf-drift.yml"
+        params {
+          ["AWS_ACCESS_KEY_ID"] = "((.:role.AWS_ACCESS_KEY_ID))"
+          ["AWS_SECRET_ACCESS_KEY"] = "((.:role.AWS_SECRET_ACCESS_KEY))"
+          ["AWS_SESSION_TOKEN"] = "((.:role.AWS_SESSION_TOKEN))"
+          ["DEPLOYMENT_PATH"] = "\(env.split("-")[0])/\(env)/environment/bastion"
+        }
+      }
+    }
+    on_failure = shared_resources.paySlackNotificationForFail("#govuk-pay-starling", "Drift detected for $BUILD_JOB_NAME <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse build #$BUILD_NAME>")
+    on_success = shared_resources.paySlackNotificationForSuccess("#govuk-pay-activity", "Terraform state consistent for $BUILD_JOB_NAME <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse build #$BUILD_NAME>")
+  }
+}
+
+  for (env in List("deploy", "staging", "production")) {
+    new {
+      name = "\(env)-check-for-zendesk-unsubscribers"
+      plan {
+        new InParallelStep {
+          in_parallel = new Listing<Step>{
+            (getStep("every-15-minutes")) { trigger = true }
+            getStep("pay-ci")
+          }
+        }
+        new TaskStep {
+          task = "assume-role"
+          file = "pay-ci/ci/tasks/assume-role.yml"
+          params {
+            when (env == "production") {
+              ["AWS_ROLE_ARN"] = "arn:aws:iam::((pay_aws_prod_account_id)):role/concourse"
+            }
+            else {
+              ["AWS_ROLE_ARN"] = "arn:aws:iam::((pay_aws_\(env)_account_id)):role/concourse"
+            }
+            ["AWS_ROLE_SESSION_NAME"] = "zendesk-unsubscriber-assume-role"
+          }
+        }
+        (loadVar("role", "assume-role/assume-role.json")) { format = "json" }
+        new TaskStep {
+          task = "check-if-unsubscribed"
+          file = "pay-ci/ci/tasks/check-sns-topic-unsubscribers.yml"
+          params {
+            ["AWS_ACCESS_KEY_ID"] = "((.:role.AWS_ACCESS_KEY_ID))"
+            ["AWS_SECRET_ACCESS_KEY"] = "((.:role.AWS_SECRET_ACCESS_KEY))"
+            ["AWS_SESSION_TOKEN"] = "((.:role.AWS_SESSION_TOKEN))"
+            ["AWS_REGION"] = "eu-west-1"
+            ["AWS_DEFAULT_REGION"] = "eu-west-1"
+            when (env=="deploy") {
+              ["REGIONS"] = "eu-west-1 eu-central-1 us-east-1"
+            }
+            else {
+              ["REGIONS"] = "eu-west-1 us-east-1"
+            }
+          }
+        }
+      }
+    on_failure = shared_resources.paySlackNotificationForFail("#govuk-pay-starling", "An SNS topic subscriber is unsubscribed - $BUILD_JOB_NAME <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse build #$BUILD_NAME>")
```

Comment:
Looking at the Concourse set-pipeline dry runs, the `on_failure` and `on_success` used by the drift detector pipeline are slightly different. I don't think we much care about the icons being changed, or the body of the message, but the username changes. Does this matter?

Reply:
The different username/icon does separate from the all deployments updates in the pay-activity channel. I would say keep the standard 'pay-concourse' username but change the message so its a little clearer what it checked